code/game/objects/effects/chem/puddle.dm

```diff
@@ -0,0 +1,109 @@
+
+/obj/effect/decal/cleanable/chempuddle
+	name = "puddle"
+	icon = 'icons/effects/chempuddle.dmi'
+	icon_state = "puddle"
+
+	mouse_opacity = FALSE
+
+/obj/effect/decal/cleanable/chempuddle/Initialize()
+	. = ..()
+
+	create_reagents(100)
+
+	spawn(10 SECONDS)
```

Comment:
It would probably be recommended to use timers rather than spawn when adding new code.